Assets/HoloToolkit/UX/Scripts/Progress/ProgressIndicator.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity;
+using UnityEngine;
+
+namespace HoloToolkit.UX.Progress
+{
+    /// <summary>
+    /// Singleton for displaying a simple loading dialog
+    /// Can be combined with a radial solver to keep locked to the HMD
+    /// </summary>
+    public class ProgressIndicator : Singleton<ProgressIndicator>
+    {
+        const float SmoothProgressSpeed = 10f;
+
+        public enum IndicatorStyleEnum
```

Comment:
If these enums are intended to be referenced outside of this class, please move them outside. if they are not, then they can be made private/protected.